Shared/Samples/Snap geometry edits/SnapGeometryEditsView.SnapSettingsView.swift

```diff
@@ -0,0 +1,85 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+extension SnapGeometryEditsView {
+    struct SnapSettingsView: View {
+        /// The action to dismiss the settings sheet.
+        @Environment(\.dismiss) private var dismiss: DismissAction
+        
+        /// The view model for the sample.
+        @ObservedObject var model: GeometryEditorModel
+        
+        /// A Boolean value indicating whether snapping is enabled
+        /// for the geometry editor.
+        @State private var snappingEnabled = false
+        
+        /// An array of snap source layer names and their source settings.
+        @State private var snapSources: [(layerName: String, sourceSettings: SnapSourceSettings)] = []
+        
+        /// An array of Boolean values for each snap source enabled states.
+        @State private var snapSourceEnabledStates: [Bool] = []
+        
+        var body: some View {
+            Form {
+                Section("Snap Source") {
+                    Toggle("Snapping", isOn: $snappingEnabled)
+                        .onChange(of: snappingEnabled) { newValue in
+                            model.geometryEditor.snapSettings.isEnabled = newValue
+                        }
+                }
+                
+                Section("Layer Snapping") {
+                    ForEach(0 ..< snapSources.count, id: \.self) { index in
+                        Toggle(snapSources[index].layerName, isOn: $snapSourceEnabledStates[index])
+                            .onChange(of: snapSourceEnabledStates[index]) { newValue in
+                                snapSources[index].sourceSettings.isEnabled = newValue
+                            }
+                    }
+                }
+            }
+            .navigationTitle("Snap Settings")
+            .navigationBarTitleDisplayMode(.inline)
+            .toolbar {
+                ToolbarItem(placement: .confirmationAction) {
+                    Button("Done") {
+                        dismiss()
+                    }
+                }
+            }
+            .onAppear {
+                // Enables snapping in the current geometry editor.
+                model.geometryEditor.snapSettings.isEnabled = true
+                snappingEnabled = model.geometryEditor.snapSettings.isEnabled
+                
+                // Creates an array from snap source layers with their
+                // layer name and source settings.
+                snapSources = model.geometryEditor.snapSettings.sourceSettings.compactMap { sourceSettings in
+                    if let layer = sourceSettings.source as? FeatureLayer {
+                        return (layer.name, sourceSettings)
+                    } else {
+                        return nil
+                    }
```

Comment:
```suggestion
                    guard let layer = sourceSettings.source as? FeatureLayer else { return nil }
                    return (layer.name, sourceSettings)
```

Not a big deal, but this seems a little cleaner to me.